pkg/util/kubernetes/autoscalers/processor.go

```diff
@@ -146,13 +149,25 @@ func (p *Processor) ProcessWPAs(wpa *v1alpha1.WatermarkPodAutoscaler) map[string
 }
 
 func makeChunks(batch []string) (chunks [][]string) {
-	for i := 0; i < len(batch); i += chunkSize {
-		if i+chunkSize > len(batch) {
-			chunks = append(chunks, batch[i:])
-			break
+	// uriLenght is used to avoid making a query that goes beyond the maximum URI size.
+	var uriLenght int
+	var tempBucket []string
+
+	for _, val := range batch {
+		// Lenght of the query plus comma, time and space aggregators that come later on.
+		tempSize := len(url.PathEscape(val)) + 16
+		uriLenght = uriLenght + tempSize
+		if uriLenght >= maxCharactersPerChunk || len(tempBucket) >= chunkSize {
```

Comment:
Would be great to extract the logic for URL checking outside of this func into a helper with semantics of `IsURLWithinLimit` and put relevant comments there.

Reply:
Can do, I will wait for a confirmation on whether we want to sort or not.
cc @clamoriniere